src/main/java/build/buildfarm/worker/Worker.java

```diff
@@ -222,17 +225,21 @@ private boolean fetchInputs(Path execDir, Digest inputRoot) throws IOException {
     ImmutableList.Builder<Directory> directories = new ImmutableList.Builder<>();
     String pageToken = "";
 
-    Map<Digest, Directory> directoriesIndex = new HashMap<Digest, Directory>();
-
     do {
       pageToken = instance.getTree(inputRoot, config.getTreePageSize(), pageToken, directories);
     } while (!pageToken.isEmpty());
 
-    directoriesIndex = Maps.uniqueIndex(
-        directories.build(),
-        directory -> Digests.computeDigest(directory));
+    Set<Digest> directoryDigests = new HashSet<>();
+    ImmutableMap.Builder<Digest, Directory> directoriesIndex = new ImmutableMap.Builder<>();
```

Comment:
What happens if you get multiple directories with the same name but different digests? Is that something we need to protect against?